csharp/CMakeLists.txt

```diff
@@ -28,185 +28,280 @@
+  csharp_OpenStudioModelSimulation_wrap.cxx
+  csharp_OpenStudioModelStraightComponent_wrap.cxx
+  csharp_OpenStudioModelZoneHVAC_wrap.cxx
+  csharp_OpenStudioOSVersion_wrap.cxx
+)
+
+set(utilities_wrappers
+  csharp_OpenStudioEnergyPlus_wrap.cxx
+  csharp_OpenStudioRadiance_wrap.cxx
+  csharp_OpenStudioGBXML_wrap.cxx
+  csharp_OpenStudioAirflow_wrap.cxx
+  csharp_OpenStudioISOModel_wrap.cxx
+  csharp_OpenStudioSDD_wrap.cxx
+)
+
+foreach(wrapper_file IN LISTS ALL_CSHARP_WRAPPER_FILES)
+  get_filename_component(wrapper_name ${wrapper_file} NAME)
+
+  set(notfound TRUE)
+
+  foreach(model_wrapper IN LISTS model_wrappers)
+    if( wrapper_name STREQUAL model_wrapper )
+      list(APPEND model_wrapper_files ${wrapper_file})
+      set(notfound FALSE)
+      break()
+    endif()
+  endforeach()
 
-  foreach(wrapper_file IN LISTS ALL_CSHARP_WRAPPER_FILES)
-    get_filename_component(wrapper_name ${wrapper_file} NAME)
-
-    set(notfound TRUE)
-
-    foreach(model_wrapper IN LISTS model_wrappers)
-      if( wrapper_name STREQUAL model_wrapper )
-        list(APPEND model_wrapper_files ${wrapper_file})
-        set(notfound FALSE)
-        break()
-      endif()
-    endforeach()
-
-    foreach(translator_wrapper IN LISTS translator_wrappers)
-      if( wrapper_name STREQUAL translator_wrapper )
-        list(APPEND translator_wrapper_files ${wrapper_file})
-        set(notfound FALSE)
-        break()
-      endif()
-    endforeach()
-
-    if( notfound )
-      list(APPEND core_wrapper_files ${wrapper_file})
+  foreach(translator_wrapper IN LISTS translator_wrappers)
+    if( wrapper_name STREQUAL translator_wrapper )
+      list(APPEND translator_wrapper_files ${wrapper_file})
+      set(notfound FALSE)
+      break()
     endif()
   endforeach()
 
-  #message("model_wrapper_files = ${model_wrapper_files}")
-  #message("translator_wrapper_files = ${translator_wrapper_files}")
-  #message("core_wrapper_files = ${core_wrapper_files}")
+  if( notfound )
+    list(APPEND core_wrapper_files ${wrapper_file})
+  endif()
+endforeach()
 
-  set_source_files_properties(${ALL_CSHARP_WRAPPER_FILES} PROPERTIES GENERATED TRUE)
+#message("model_wrapper_files = ${model_wrapper_files}")
+#message("translator_wrapper_files = ${translator_wrapper_files}")
+#message("core_wrapper_files = ${core_wrapper_files}")
 
-  # build dlls to link all the static C# bindings with
+set_source_files_properties(${ALL_CSHARP_WRAPPER_FILES} PROPERTIES GENERATED TRUE)
 
-  add_library(
-    openstudio_csharp
-    MODULE
-    ${core_wrapper_files}
-  )
-  add_dependencies(openstudio_csharp ${ALL_CSHARP_WRAPPER_TARGETS})
-  target_link_libraries(openstudio_csharp
-   openstudiolib
-  )
-  install(TARGETS openstudio_csharp DESTINATION CSharp/openstudio/ CONFIGURATIONS DEBUG COMPONENT "CSharpAPI")
-  install(TARGETS openstudio_csharp DESTINATION CSharp/openstudio/ CONFIGURATIONS RELEASE COMPONENT "CSharpAPI")
+add_library(
+  openstudio_csharp
+  SHARED # Was "MODULE" on Windows before
```

Comment:
Can't remember why I had to change that to "SHARED" instead of "MODULE", not 100% sure the impact it'll have.